pkg/cluster/cluster.go

```diff
@@ -294,6 +294,10 @@ func (c *Cluster) Create() error {
 		c.eventRecorder.Eventf(c.GetReference(), v1.EventTypeNormal, "Services", "The service %q for role %s has been successfully created", util.NameFromMeta(service.ObjectMeta), role)
 	}
 
+	if err = c.readValidateDatabaseNameRegexp(c.OpConfig.DatabaseNameRegexp); err != nil {
+		return err
+	}
+
```

Comment:
It should rather place the [existing validation](https://github.com/zalando/postgres-operator/blob/master/pkg/cluster/database.go#L419).